apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -316,35 +314,27 @@ def render(element_html: str, data: pl.QuestionData) -> str:
 
         html_params["feedback"] = partial_score.get("feedback", None)
 
-        with open(NUMBER_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
-            return chevron.render(f, html_params).strip()
+        return chevron.render(template, html_params).strip()
 
     elif data["panel"] == "answer":
-        ans_true = None
-        if pl.get_boolean_attrib(
-            element, "show-correct-answer", SHOW_CORRECT_ANSWER_DEFAULT
-        ):
-            ans_true = format_true_ans(element, data, name)
-
-        if ans_true is None:
-            return ""
-
         html_params = {
             "answer": True,
             "label": label,
-            "a_tru": ans_true,
             "suffix": suffix,
         }
-        with open(NUMBER_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
-            return chevron.render(f, html_params).strip()
+
+        if show_correct_answer:
```

Comment:
Isn't the same logic as before. If this is false, the panel shouldn't be rendered at all (i.e. return the empty string).

Reply:
Changed in the latest commit https://github.com/PrairieLearn/PrairieLearn/pull/8435/commits/1e62938d8be083b420f3e86668e0dbc845573c2e.

Reply:
@Frozemint as written, this could still have weird behavior, since there may be parts of the template that get rendered even on an empty dictionary. It would be better to explicitly return the empty string if `show_correct_answer` is false, since this results in fewer calls to chevron and is easier to read (along with leading to less deeply nested code).

Reply:
Actually, now that we're at it, do we want to return the empty string, or a slightly more informative message such as "The correct answers are hidden for this question"?

Reply:
This logic right now is written to be consistent with other elements, so if we add a message like that I think we would want to do it everywhere. But I'm not sure there's a solid use case for this, since this is something that has to be manually specified by the instructor, and in those cases the instructor should mention something about the correct answer being hidden.

Reply:
That makes sense. IIRC there is a feature for instructors to manually specify what goes into the answer panel (pl-answer-panel or something?), so I guess if the instructor sets show-correct-answer to false they can also set the answer panel. So yeah, defaulting to empty string sounds good. 

Noting this here as note to self and for future reference: I still think it's not ideal and could be confusing for students to default to a blank answer panel if answers are hidden, and perhaps we should consider hiding the entire answer panel altogether to avoid confusion. But that sounds like a nontrivial change and work for another issue/PR.

Reply:
There actually is already a way to hide the answer panel completely! It has to be configured as part of the info.json for a question.